hcatalog/core/src/test/java/org/apache/hive/hcatalog/mapreduce/TestHCatExtension.java

```diff
@@ -0,0 +1,111 @@
+import org.apache.hive.hcatalog.data.DefaultHCatRecord;
+import org.apache.hive.hcatalog.data.HCatRecord;
+import org.apache.hive.hcatalog.data.schema.HCatFieldSchema;
+import org.apache.hive.hcatalog.data.schema.HCatSchemaUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class TestHCatExtension extends HCatMapReduceTest {
+    private static List<HCatRecord> writeRecords;
+    private static List<HCatFieldSchema> dataColumns;
+    protected static final int NUM_RECORDS = 20;
+    protected static final int NUM_TOP_PARTITIONS = 5;
+
+    public TestHCatExtension(String formatName, String serdeClass, String inputFormatClass, String outputFormatClass) throws Exception {
+        super(formatName, serdeClass, inputFormatClass, outputFormatClass);
+        tableName = "testHCatExtension_" + formatName;
+        generateWriteRecords(NUM_RECORDS, NUM_TOP_PARTITIONS, 0);
+        generateDataColumns();
+    }
+
+    protected static void generateDataColumns() throws HCatException {
+        dataColumns = new ArrayList<HCatFieldSchema>();
+        dataColumns.add(HCatSchemaUtils.getHCatFieldSchema(new FieldSchema("c1", serdeConstants.INT_TYPE_NAME, "")));
+        dataColumns.add(HCatSchemaUtils.getHCatFieldSchema(new FieldSchema("c2", serdeConstants.STRING_TYPE_NAME, "")));
+        dataColumns.add(HCatSchemaUtils.getHCatFieldSchema(new FieldSchema("p1", serdeConstants.STRING_TYPE_NAME, "")));
+        dataColumns.add(HCatSchemaUtils.getHCatFieldSchema(new FieldSchema("p2", serdeConstants.STRING_TYPE_NAME, "")));
+
+    }
+
+    protected static void generateWriteRecords(int max, int mod, int offset) {
+        writeRecords = new ArrayList<HCatRecord>();
+
+        for (int i = 0; i < max; i++) {
+            List<Object> objList = new ArrayList<Object>();
+
+            objList.add(i);
+            objList.add("strvalue" + i);
+            objList.add(String.valueOf((i % mod) + offset));
+            objList.add(String.valueOf((i / (max / 2)) + offset));
+            writeRecords.add(new DefaultHCatRecord(objList));
+        }
+    }
+
+    @Override
+    protected List<FieldSchema> getPartitionKeys() {
+        List<FieldSchema> fields = new ArrayList<FieldSchema>();
+        fields.add(new FieldSchema("p1", serdeConstants.STRING_TYPE_NAME, ""));
+        fields.add(new FieldSchema("p2", serdeConstants.STRING_TYPE_NAME, ""));
+        return fields;
+    }
+
+    @Override
+    protected List<FieldSchema> getTableColumns() {
+        List<FieldSchema> fields = new ArrayList<FieldSchema>();
+        fields.add(new FieldSchema("c1", serdeConstants.INT_TYPE_NAME, ""));
+        fields.add(new FieldSchema("c2", serdeConstants.STRING_TYPE_NAME, ""));
+        return fields;
+    }
+
+    @Test
+    public void testHCatExtension() throws Exception {
+        HashMap<String, String> properties = new HashMap<String, String>();
```

Comment:
Can be simplified as `Map<String, String> properties = new HashMap<>();`

Reply:
Changed.